api/opencl_runtime_layer.asciidoc

```diff
@@ -2043,9 +2043,8 @@ Kernel Read Or Write>> table.
     objects.
 
 10::
-    sRGB channel order support is not required for 1D image buffers.
-    Writes to images with sRGB channel orders requires device support of the
-    *cl_khr_srgb_image_writes* extension.
+    Read sRGB channel order support is optional for 1D image buffers.
+    Write sRGB channel order support is optional for all image types.
```

Comment:
I'm wondering about this footnote change.  It effectively promotes an extension to optional-core.  So previously the absense `cl_khr_srgb_image_writes` would tell the programmer something, but now an implementation could report support without reporting the extension.

If any code out their checks for the extension before using sRGB writes, and  a future implementation supports sRGB writes without reporting the extension then the "wrong" path will be taken.   The cross-product of those two "if" may be extremely small and I may be overthinking this.  I have no real examples of why this would matter.

I also can't think why sRGB writes would have been completely excluded before.   If this footnote had used your proposed wording all along it would make more sense.

Reply:
I've made this comment because I followed your notes in https://github.com/KhronosGroup/OpenCL-Docs/issues/20, but I'm just want to confirm that we're okay promoting `cl_khr_srgb_image_writes` during maintenance.

Reply:
Yeah, I remain confused by this extension. :smiley:

I believe the scenario we are discussing is:

* An implementation supports writes to sRGB images by including them in the set of writeable image formats returned by `clGetSupportedImageFormats`, but...
* The implementation does not report support for `cl_khr_srgb_image_writes`.

Is this valid or invalid?

Said another way, does an implementation **have to** support the `cl_khr_srgb_image_writes` extension if it supports writing to sRGB images?

I will check the CTS tests to see if it checks for this condition.

Reply:
We discussed this on the teleconference, essentially if the image type is in the list of supported images (which it would be with this extension, and may be without) then it's fine.  Resolving.